docs: Add troubleshooting section on checkpointing for EventHub Azure Functions using the new EventHub functions extension. (#686)
kedacore/keda-docs: content/docs/2.3/scalers/azure-event-hub.md

When you do so, the Event Hub scaler will depend on the existence of two configurations you have to provide: `eventHubNamespace` and `eventHubName`.

### Checkpointing Behaviour {#checkpointing-behaviour}

> * **Legacy behaviour:** C# applications, which use the `Microsoft.Azure.EventHubs` package, Java applications which use the `azure-eventhubs-eph` package or Python applications which use the `azure-eventhub` below `v5` are supported if no `checkpointStrategy` is specified. These legacy implementations are based on the `EventProcessorHost` client, which stores checkpoint information as blob container content.  

Reviewer: Three things I would suggest is:
1. Add link to the values that you can specify in `checkpointStrategy` as we don't talk about legacy & current there
2. Move the Frameworks and respective packages to a list so that it's easier to find, maybe? (example below)
3. Cover remaining strategies (OK in separate PR if you prefer to)

```suggestion
* **Legacy behaviour:** Checkpoint information is stored as blob container. Default behavior when no `checkpointStrategy` is specified.
  - .NET applications using `Microsoft.Azure.EventHubs` NuGet package
  - Java  applications using `azure-eventhubs-eph` {?} package
  - Python applications using `azure-eventhub` {?} package below 5.0
```

Author: Could you be a bit more specific on what points should be covered for the remaining strategies?

Reviewer: With 3) I was more referring to this:
![image](https://user-images.githubusercontent.com/4345663/157211202-467b8f2f-fa48-45db-94cf-7b161b13d352.png)

Where we currently talk about legacy and current, but what about the other options for `checkpointStrategy` but maybe I'm overthikning things @kedacore/keda-maintainers?

Author: I am not yet sure on what I should be adding besides the information already mentioned under the parameters section. Happy to raise a separate PR for it, if someone clarifies that to me.

Reviewer: I've had another look and let's just leave it as-is, we can still improve if need be.

When you do so, the Event Hub scaler will depend on the existence of two configurations you have to provide: `eventHubNamespace` and `eventHubName`.

### Checkpointing Behaviour

The list of available checkpointing strategies can be found in the trigger specification [section](#trigger-specification). The way checkpoints are stored
has changed with updates to the EventHub SDKs.

* **Legacy behaviour:** The older implementations are based on the `EventProcessorHost` client, which stores the checkpoint information as contents of
the storage blob. This is the default behaviour when no `checkpointStrategy` is specified.

> The following applications will showcase this -

Reviewer: This might be better?
```suggestion
This is applicable for the following scenarios:
```

  - .NET applications using `Microsoft.Azure.EventHubs` NuGet package.
  - Java applications using `azure-eventhubs-eph` package.
  - Python applications using `azure-eventhub` package below v5.

* **Current behaviour:** The newer implementations are based on the `EventProcessorClient`, which stores the checkpoint information as metadata on
the storage blob. This is the behaviour when `checkpointStrategy` is set to `blobMetadata`.

> The following applications will showcase this -

Reviewer: This might be better?
```suggestion
This is applicable for the following scenarios:
```

Author: Done.
